bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -372,68 +258,84 @@ private void updateGuiding(final TargetEnvironment env) {
         // Update the guiding feedback.
         final ISPObsComponent node = getContextTargetObsComp();
         final SPTarget      target = TargetSelection.get(env, node);
-        _w.detailEditor.gfe().edit(getObsContext(env), target, node);
+        _w.detailEditor.gfe().edit(ctx, target, node);
     }
 
-    private final PropertyChangeListener selectionListener = new PropertyChangeListener() {
-        @Override
-        public void propertyChange(PropertyChangeEvent evt) {
-            final ISPObsComponent node = getContextTargetObsComp();
-            final TargetEnvironment env = getDataObject().getTargetEnvironment();
-            final SPTarget target = TargetSelection.get(env, node);
-            if (target != null) {
-                if (getDataObject() != null) {
-                    final TargetEnvironment env1 = getDataObject().getTargetEnvironment();
-                    if (env1.getTargets().contains(target)) {
-
-                        if (_curPos != null) _curPos.deleteWatcher(posWatcher);
-                        _curPos = target;
-                        _curPos.addWatcher(posWatcher);
-                        refreshAll();
-                        updateRemovePrimaryButtonsAndDetailEditor(env1);
-                    }
-                }
-            } else {
-                final GuideGroup grp = TargetSelection.getGuideGroup(env, node);
-                if (grp != null) if (getDataObject() != null) {
-                    final TargetEnvironment env1 = getDataObject().getTargetEnvironment();
-                    if (env1.getGroups().contains(grp)) {
 
-                        if (_curPos != null) _curPos.deleteWatcher(posWatcher);
+    private static boolean enablePrimary(final SPTarget target, final TargetEnvironment env) {
+        return env.getBase() != target && !env.getUserTargets().contains(target);
```

Comment:
In all of these cases, we actually DO want `==` or `!=`, as we are working directly with references to the same objects. If we used `equals` instead, we would end up having to compare watcher lists on the `SPTarget`s and the underlying `ITarget`s, which is computationally unnecessary, and indeed, it is even possible it could result in false positives for two freshly created `SPTargets` which would have identical freshly created `ITarget`s with regards to `equals`.

Absolutely contrary to programmer training and intuition :smile: